webapp/packages/core-authentication/src/PasswordPolicyService.ts

```diff
@@ -25,22 +25,25 @@ type ValidationResult = { isValid: true; errorMessage: null } | { isValid: false
 export class PasswordPolicyService {
   get config(): PasswordPolicyConfig {
     return {
-      minLength: this.serverConfigResource.data?.passwordPolicyConfiguration?.minLength || DEFAULT_PASSWORD_POLICY.minLength,
-      minNumberCount: this.serverConfigResource.data?.passwordPolicyConfiguration?.minNumberCount || DEFAULT_PASSWORD_POLICY.minNumberCount,
-      minSymbolCount: this.serverConfigResource.data?.passwordPolicyConfiguration?.minSymbolCount || DEFAULT_PASSWORD_POLICY.minSymbolCount,
-      requireMixedCase: this.serverConfigResource.data?.passwordPolicyConfiguration?.requireMixedCase || DEFAULT_PASSWORD_POLICY.requireMixedCase,
+      minLength: this.passwordPolicyResource.data?.minLength || DEFAULT_PASSWORD_POLICY.minLength,
+      minNumberCount: this.passwordPolicyResource.data?.minNumberCount || DEFAULT_PASSWORD_POLICY.minNumberCount,
+      minSymbolCount: this.passwordPolicyResource.data?.minSymbolCount || DEFAULT_PASSWORD_POLICY.minSymbolCount,
+      requireMixedCase: this.passwordPolicyResource.data?.requireMixedCase || DEFAULT_PASSWORD_POLICY.requireMixedCase,
     };
   }
 
   constructor(
-    private readonly serverConfigResource: ServerConfigResource,
+    private readonly passwordPolicyResource: PasswordPolicyResource,
     private readonly localizationService: LocalizationService,
   ) {
     makeObservable(this, {
       config: computed,
     });
   }
 
+  /**
+   * PasswordPolicyResource should be loaded before calling this method
+   */
   validatePassword(password: string): ValidationResult {
```

Comment:
I believe we should add a check here. There is a chance a person would not have read the comment and used this function without the resource had been loaded before

```suggestion
  validatePassword(password: string): ValidationResult {
    if (!this.passwordPolicyResource.isLoaded()) {
      throw new Error('Password policy is not loaded yet');
    }
```

Reply:
it can be outdated but loaded, so we need to know that we need to manage it at place